File: Assets/PolitiRush/Core/Weapon.cs
namespace PolitiRush.Core
{
    /// <summary>Arme du héros, déduite du multiplicateur de tir. Miroir de Weapon.kt.</summary>
    public enum Weapon { Main, LancePierre, Canon, Tank }

    public static class WeaponInfo
    {
        public static string Label(this Weapon w) => w switch
        {
            Weapon.Main => "À la main", Weapon.LancePierre => "Lance-pierre", Weapon.Canon => "Canon à tomates", _ => "Tank à tomates",
        };

        public static int MinFirepower(this Weapon w) => w switch { Weapon.Main => 1, Weapon.LancePierre => 5, Weapon.Canon => 13, _ => 33 };

        public static Weapon ForFirepower(int firepower)
        {
            if (firepower >= Weapon.Tank.MinFirepower()) return Weapon.Tank;
            if (firepower >= Weapon.Canon.MinFirepower()) return Weapon.Canon;
            if (firepower >= Weapon.LancePierre.MinFirepower()) return Weapon.LancePierre;
            return Weapon.Main;
        }
    }
}

File: core/src/main/kotlin/com/sealkrach/politirush/core/Weapon.kt
package com.sealkrach.politirush.core

/**
 * Arme du héros, déduite du multiplicateur de tir. Purement visuelle et
 * narrative : la mécanique reste "firepower projectiles par salve", mais le
 * joueur voit son équipement grossir avec sa puissance.
 */
enum class Weapon(val label: String, val minFirepower: Int) {
    MAIN("À la main", 1),
    LANCE_PIERRE("Lance-pierre", 5),
    CANON("Canon à tomates", 13),
    TANK("Tank à tomates", 33);

    companion object {
        fun forFirepower(firepower: Int): Weapon =
            entries.last { firepower >= it.minFirepower }
    }
}
